index.coffee

```diff
@@ -19,16 +20,25 @@ tryExtractDigestHash = (evt) ->
 		matchPull = evt.status.match(/^Digest:\s([a-zA-Z0-9]+:[a-f0-9]+)$/)
 		return matchPull[1] if matchPull?
 
-# Extracts the digest value of an image from docker events
-# for push and pull operations, if no digest value is found
-# null is returned
-extractDigestHash = (stream) ->
-	# iterator over the event stream in reverse order
-	# the digest event is one of the last ones.
-	for idx in [stream.length - 1..0] by -1
-		hash = tryExtractDigestHash(stream[idx])
-		return hash if hash?
-	return null
+awaitRegistryStream = (stream, onProgress) ->
+	contentHash = null
+	new Promise (resolve, reject) ->
+
+		jsonStream = JSONStream.parse()
+
+		jsonStream.on 'data', (evt) ->
+			if typeof evt isnt 'object'
+				onProgress({})
```

Comment:
No need to call the handler in this case, in fact I think it might mess things up for some clients if it's an empty object.

Reply:
I was keeping the iterface the same as `followProgress`, which does this: https://github.com/apocas/docker-modem/blob/master/lib/modem.js#L323

Reply:
I see, that's fine then -- it seems clients already need to handle the empty object so we're good.